drivers/wifi/nxp/nxp_wifi_drv.c

```diff
@@ -0,0 +1,1842 @@
+		if (params->scan_type == WIFI_SCAN_TYPE_ACTIVE) {
+			wlan_scan_params_v2.chan_list[i].scan_type = MLAN_SCAN_TYPE_ACTIVE;
+			wlan_scan_params_v2.chan_list[i].scan_time = params->dwell_time_active;
+		} else {
+			wlan_scan_params_v2.chan_list[i].scan_type = MLAN_SCAN_TYPE_PASSIVE;
+			wlan_scan_params_v2.chan_list[i].scan_time = params->dwell_time_passive;
+		}
+
+		wlan_scan_params_v2.chan_list[i].chan_number =
+			(uint8_t)(params->band_chan[i].channel);
+	}
+#endif
+
+	wlan_scan_params_v2.num_channels = i;
+
+	if (params->bands & (1 << WIFI_FREQ_BAND_2_4_GHZ)) {
+		wlan_scan_params_v2.chan_list[0].radio_type = 0 | BAND_SPECIFIED;
+	}
+#ifdef CONFIG_5GHz_SUPPORT
+	if (params->bands & (1 << WIFI_FREQ_BAND_5_GHZ)) {
+		if (wlan_scan_params_v2.chan_list[0].radio_type & BAND_SPECIFIED) {
+			wlan_scan_params_v2.chan_list[0].radio_type = 0;
+		} else {
+			wlan_scan_params_v2.chan_list[0].radio_type = 1 | BAND_SPECIFIED;
+		}
+	}
+#else
+	if (params->bands & (1 << WIFI_FREQ_BAND_5_GHZ)) {
+		LOG_ERR("Wi-Fi band 5Hz not supported");
+		g_mlan.scan_cb = NULL;
+		return -EIO;
+	}
+#endif
+
+do_scan:
+	wlan_scan_params_v2.cb = nxp_wifi_process_results;
+
+	ret = wlan_scan_with_opt(wlan_scan_params_v2);
+	if (ret != WM_SUCCESS) {
+		LOG_ERR("Failed to start Wi-Fi scanning");
+		g_mlan.scan_cb = NULL;
+		return -EAGAIN;
+	}
+
+	return 0;
+}
+
+static int nxp_wifi_version(const struct device *dev, struct wifi_version *params)
+{
+	nxp_wifi_ret_t status = NXP_WIFI_RET_SUCCESS;
+
+	if (s_nxp_wifi_State != NXP_WIFI_STARTED) {
+		status = NXP_WIFI_RET_NOT_READY;
+	}
+
+	if (status != NXP_WIFI_RET_SUCCESS) {
+		LOG_ERR("Failed to get Wi-Fi driver/firmware version");
+		return -EAGAIN;
+	}
+
+	params->drv_version = WLAN_DRV_VERSION;
+
+	params->fw_version = wlan_get_firmware_version_ext();
+
+	return 0;
+}
+
+static int nxp_wifi_ap_bandwidth(const struct device *dev, struct wifi_ap_params *params)
+{
+	nxp_wifi_ret_t status = NXP_WIFI_RET_SUCCESS;
+	int ret = WM_SUCCESS;
+	interface_t *if_handle = (interface_t *)dev->data;
+
+	if (if_handle->state.interface != WLAN_BSS_TYPE_UAP) {
+		LOG_ERR("Wi-Fi not in uAP mode");
+		return -EIO;
+	}
+
+	if (s_nxp_wifi_State != NXP_WIFI_STARTED) {
+		status = NXP_WIFI_RET_NOT_READY;
+	}
+
+	if (status == NXP_WIFI_RET_SUCCESS) {
+
+		if (params->oper == WIFI_MGMT_SET) {
+
+			ret = wlan_uap_set_bandwidth(params->bandwidth);
+
+			if (ret != WM_SUCCESS) {
+				status = NXP_WIFI_RET_FAIL;
+			}
+		} else {
+			ret = wlan_uap_get_bandwidth(&params->bandwidth);
+
+			if (ret != WM_SUCCESS) {
+				status = NXP_WIFI_RET_FAIL;
+			}
+		}
+	}
+
+	if (status != NXP_WIFI_RET_SUCCESS) {
+		LOG_ERR("Failed to get/set Wi-Fi AP bandwidth");
+		return -EAGAIN;
+	}
+
+	return 0;
+}
+
+static int nxp_wifi_connect(const struct device *dev, struct wifi_connect_req_params *params)
+{
+	nxp_wifi_ret_t status = NXP_WIFI_RET_SUCCESS;
+	int ret;
+	interface_t *if_handle = (interface_t *)dev->data;
```

Comment:
Usually we only use new types for function pointers in zephyr, I suggest this driver follows this rule. So here it would be better just have `struct xyz *if_handle` or similar.